### PR TITLE
fix: enabled toggle behavior for settings icon in sidebar

### PR DESCRIPTION
**Pull request by saiprabhu-dandanayak**
### Bug : [Settings Bug Link](https://github.com/appsmithorg/appsmith/issues/32030)

I have raised this PR In-order to ensure that the App Settings pane is closed when clicked on settings icon when it is open

### Cypress testing video

https://github.com/appsmithorg/appsmith/assets/122865888/cd8b3452-3776-4ac2-82a0-577a7f9b1065

## Summary by CodeRabbit

- **Tests**
  - Refactored tests for table widget column editability to use a new method for toggling column settings.
  - Updated import paths and function calls in multiple test files for better modularity.
  - Enhanced test logic for SidebarButton to ensure `onClick` is triggered when toggling state.
  - Introduced a new test suite for validating the functionality of the App Settings pane, ensuring it behaves correctly when the settings icon is clicked.
- **Chores**
  - Removed unnecessary properties and adjusted string messages in JSON fixtures to improve clarity and alignment with actual behavior.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/AppNavigation/AppNavigation_Toggle_icon_Spec.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import {
```
Comment (rajatagrawal) [inline app/client/cypress/e2e/Regression/ClientSide/AppNavigation/AppNavigation_Toggle_icon_Spec.js:1]: Can we please test this using unit tests and react testing library please ?

cc : @ApekshaBhosale @sagar-qa007 @yatinappsmith

**Review by rajatagrawal — CHANGES_REQUESTED** [review]
Left comments

File: app/client/cypress/e2e/Regression/ClientSide/AppNavigation/AppNavigation_Toggle_icon_Spec.js (hunk 1 above)
Comment (saiprabhu-dandanayak) [reply app/client/cypress/e2e/Regression/ClientSide/AppNavigation/AppNavigation_Toggle_icon_Spec.js:1]: Hi **@rajatagrawal** , I've just pulled the latest changes, and I noticed that the **sidebar** folder has been deleted. The files within it have been segregated into separate directories. Consequently, I've raised an updated pull request (PR) to reflect these changes.

**Comment by ayushpahwa** [comment]
Updated reviewers based on current context.

**Comment by hetunandu** [comment]
This issue should not be worked on as it requests a user experience that does not currently align to the IDE usability. Closing this

**Pull request closed without merging**